File: src/clients/deno/examples/window-size.ts
import { createWebView } from "../main.ts";

using webview = await createWebView({
  title: "Window Size",
  html: `
    <h1>Window Sizes</h1>
    <div style="display: flex; gap: 10px;">
      <button onclick="window.ipc.postMessage('maximize')">Maximize</button>
      <button onclick="window.ipc.postMessage('minimize')">Minimize</button>
      <button onclick="window.ipc.postMessage('fullscreen')">Fullscreen</button>
    </div>
  `,
  size: {
    height: 200,
    width: 800,
  },
  ipc: true,
});

webview.on("ipc", ({ message }) => {
  switch (message) {
    case "maximize":
      webview.maximize();
      break;
    case "minimize":
      webview.minimize();
      break;
    case "fullscreen":
      webview.fullscreen();
      break;
    default:
      console.error("Unknown message", message);
  }
});

await webview.waitUntilClosed();

File: src/clients/python/examples/window_size.py
# /// script
# requires-python = ">=3.13"
# dependencies = [
#     "justbe-webview",
# ]
#
# [tool.uv.sources]
# justbe-webview = { path = "../" }
# ///
import asyncio

from justbe_webview import WebView, Options, ContentHtml, IpcNotification, Size


async def main():
    print("Creating webview")
    config = Options(
        title="Window Size",
        load=ContentHtml(
            html="""
            <h1>Window Sizes</h1>
            <div style="display: flex; gap: 10px;">
                <button onclick="window.ipc.postMessage('maximize')">Maximize</button>
                <button onclick="window.ipc.postMessage('minimize')">Minimize</button>
                <button onclick="window.ipc.postMessage('fullscreen')">Fullscreen</button>
            </div>
            """
        ),
        size=Size(width=800, height=200),
        ipc=True,
    )

    async with WebView(config) as webview:

        async def handle_ipc(event: IpcNotification):
            message = event.message
            if message == "maximize":
                await webview.maximize()
            elif message == "minimize":
                await webview.minimize()
            elif message == "fullscreen":
                await webview.fullscreen()
            else:
                print(f"Unknown message: {message}")

        webview.on("ipc", handle_ipc)

    print("Webview closed")


if __name__ == "__main__":
    asyncio.run(main())
